python/res/enkf/jobs/enkf_scaling_job.py

```diff
@@ -0,0 +1,220 @@
+    else:
+        return observation_data.copy_active_list(key).setParent()
+
+
+def _make_tuple(key, index, active_list):
+    new_event = namedtuple("named_dict", ["key", "index", "active_list"])
+    return new_event(key, index, active_list)
+
+
+def _update_scaling(obs, scale_factor, events):
+    """
+    Applies the scaling factor to the user specified index, SUMMARY_OBS needs to be treated differently
+    as it only has one data point per node, compared with other observation types which have multiple
+    data points per node.
+    """
+    for event in events:
+        obs_vector = obs[event.key]
+        for index, obs_node in enumerate(obs_vector):
+            if obs_vector.getImplementationType().name == "SUMMARY_OBS":
+                index_list = event.index if event.index is not None else range(len(obs_vector))
+                if index in index_list:
+                    obs_node.set_std_scaling(scale_factor)
+            elif obs_vector.getImplementationType().name != "SUMMARY_OBS":
+                obs_node.updateStdScaling(scale_factor, event.active_list)
+
+
+def _active_list_from_index_list(index_list):
+    """
+    Creates an ActiveList from a list of indexes
+    :param index_list: list of index
+    :type index_list:  list
+    :return: Active list, a c-object with mode (ALL-ACTIVE, PARTIALLY-ACTIVE, INACTIVE) and list of indices
+    :rtype: active_list
+    """
+    active_list = ActiveList()
+    [active_list.addActiveIndex(index) for index in index_list]
+    return active_list
+
+
+def _set_active_lists(observation_data, key_list, active_lists):
+    """
+    Will make a backup of the existing active list on the observation node
+    before setting the user supplied index list.
+    """
+    exisiting_active_lists = []
+    for key, active_list in zip(key_list, active_lists):
+        exisiting_active_lists.append(observation_data.copy_active_list(key))
+        observation_data.setActiveList(key, active_list)
+    return observation_data, exisiting_active_lists
+
+
+def valid_job(observations, user_config):
+    """
+    Validates the job, assumes that the configuration is valid
+    """
+
+    calculation_keys = user_config.snapshot.CALCULATE_KEYS.keys
+    application_keys = [entry.key for entry in user_config.snapshot.UPDATE_KEYS.keys]
+
+    subset = is_subset(calculation_keys, application_keys)
+    obs_keys_present = has_keys(observations, calculation_keys)
+
+    return all([subset, obs_keys_present])
+
+
+def has_keys(observations, keys):
+    """
+    Checks that all keys are present in the observations
+    """
+    return all(key in observations for key in keys)
+
+
+def is_subset(main_list, sub_list):
+    """
+    Checks if all the keys in sub_list are present in main_list
+    """
+    return set(sub_list).issubset(set(main_list))
+
+
+def valid_configuration(user_config):
+    """
+    Validates the configuration
+    """
+
+    for error in user_config.errors:
+        print(error)
```

Comment:
The errors should be written to `stderr` as a non-zero `sys.exit` follows.

Reply:
Rewrote the error handling for the config and the job slightly and incorporated writing to `stderr`